test/functional/tests/ci/test_ci_write_mode_read_only.py

```diff
@@ -0,0 +1,187 @@
+#
+# Copyright(c) 2022 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+#
+
+import pytest
+import time
+
+from api.cas import casadm
+from api.cas.cli import casadm_bin
+from core.test_run import TestRun
+from storage_devices.disk import DiskType, DiskTypeSet, DiskTypeLowerThan
+from test_utils.size import Unit, Size
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.nand, DiskType.optane]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+@pytest.mark.CI()
+def test_ci_write_mode():
+    """
+    title: Verification test for write mode: read only
+    description: Verify if write mode: read only, works as expected and cache only reads
+    and do not cache writes
+    pass criteria:
+    - cache does not cache write
+    """
+
+    with TestRun.step("Prepare devices"):
+        cache_device = TestRun.disks["cache"]
+        core_device = TestRun.disks["core"]
+
+        cache_device.create_partitions([Size(1024, Unit.MebiByte)])
+        core_device.create_partitions([Size(2048, Unit.MebiByte)])
+
+        cache_device = cache_device.partitions[0]
+        core_device = core_device.partitions[0]
+
+    with TestRun.step("Test read from cache"):
+        count = 100
+        block_size = 4096
+        iostat_block_size = 1024
+        modes = ["wt", "wa"]
+
+        for mode in modes:
+            print("------Start CAS Linux in " + mode + " mode")
+            # This is where the real test starts
+            cache = casadm.start_cache(cache_dev=cache_device, cache_id=1, force=True)
+            casadm.add_core(cache, core_device)
+            time.sleep(2.5)
+            TestRun.executor.run_expect_success(f"{casadm_bin} -Q -i 1 -c {mode}")
+            TestRun.executor.run_expect_success(
+                f"dd if=/dev/cas1-1 of=$TMP_DIR/file001.bin bs=4096 count={count} "
+                f"skip=10000 iflag=direct")
+            temp = "{print $5}"
+            tmp_cache = cache_device.parent_device.unplug_command
+            tmp_cache = tmp_cache.split('/')
+            tmp_cache = tmp_cache[3]
+            read_cache_1 = TestRun.executor.run_expect_success(
+                f"iostat {cache_device.parent_device.path} | grep {tmp_cache} | "
+                f"awk 'NR==1 {temp}'").stdout
+            TestRun.executor.run_expect_success(
+                f"dd if=/dev/cas1-1 of=$TMP_DIR/file001.bin bs=4096 count={count} "
+                f"skip=10000 iflag=direct")
+            read_cache_2 = TestRun.executor.run_expect_success(
+                f"iostat {cache_device.parent_device.path} | grep {tmp_cache} | "
+                f"awk 'NR==1 {temp}'").stdout
```

Comment:
We have wrappers for casadm, dd, iostat - why not use them?
casadm.set_cache_mode(), test_tools/dd.py, test_tools/iostat.py